distributed/tests/test_nanny.py

```diff
@@ -610,3 +611,21 @@ async def test_environ_plugin(c, s, a, b):
         assert results[a.worker_address] == "123"
         assert results[b.worker_address] == "123"
         assert results[n.worker_address] == "123"
+
+
+@pytest.mark.xfail(reason="https://github.com/dask/distributed/issues/5723")
+@gen_cluster(client=True, Worker=Nanny, nthreads=[("", 1)])
+async def test_no_unnecessary_imports_on_worker(c, s, a):
```

Comment:
```suggestion
@pytest.mark.parametrize("modname", ["numpy", "pandas"])
@gen_cluster(client=True, Worker=Nanny, nthreads=[("", 1)])
async def test_no_unnecessary_imports_on_worker(c, s, a, modname):
```

Reply:
NumPy currently seems to always be imported (AFAICT locally). I dropped a breakpoint in `numpy/__init__.py` and got:
```
-> import distributed
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/distributed/distributed/__init__.py(7)<module>()
-> from .actor import Actor, ActorFuture
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/distributed/distributed/actor.py(5)<module>()
-> from .client import Future
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/distributed/distributed/client.py(59)<module>()
-> from .batched import BatchedSend
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/distributed/distributed/batched.py(10)<module>()
-> from .core import CommClosedError
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/distributed/distributed/core.py(27)<module>()
-> from . import profile, protocol
  <frozen importlib._bootstrap>(1058)_handle_fromlist()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/distributed/distributed/protocol/__init__.py(4)<module>()
-> from .compression import compressions, default_compression
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/distributed/distributed/protocol/compression.py(20)<module>()
-> import blosc
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/dask-playground/env/lib/python3.9/site-packages/blosc/__init__.py(85)<module>()
-> from blosc.test import run as test
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
  /Users/gabe/dev/dask-playground/env/lib/python3.9/site-packages/blosc/test.py(15)<module>()
-> import numpy
  <frozen importlib._bootstrap>(1007)_find_and_load()
  <frozen importlib._bootstrap>(986)_find_and_load_unlocked()
  <frozen importlib._bootstrap>(680)_load_unlocked()
  <frozen importlib._bootstrap_external>(790)exec_module()
  <frozen importlib._bootstrap>(228)_call_with_frames_removed()
> /Users/gabe/dev/dask-playground/env/lib/python3.9/site-packages/numpy/__init__.py(110)<module>()
```

tl;dr: compression tries to import blosc, blosc imports its own testing suite, that test suite imports numpy.

If I remove blosc (`sys.modules["blosc"] = None`), and use this PR plus https://github.com/dask/distributed/pull/5724, I can start a Nanny worker without import NumPy.